plugins/fs/guest-js/index.ts

```diff
@@ -1009,6 +1009,10 @@ async function writeFile(
     throw new TypeError("Must be a file URL.");
   }
 
+  if (!options || !options.baseDir) {
+    path = path instanceof URL ? path : new URL(encodeURI(path));
```

Comment:
This will throw if provided path is something like this `./file` or just `file`, because `new URL()` expects a valid URL e.g. `<scheme>://<domain>`

Reply:
I'm aware of this issue and appreciate you pointing it out. To resolve this, I decided to abandon using `new URL()` and instead used `TextEncoder`, adding decoding logic on the Rust side.